File: app/src/main/java/com/alphadevelopmentsolutions/frcscout/Enums/GameScoreStatus.java
package com.alphadevelopmentsolutions.frcscout.Enums;

public enum  GameScoreStatus
{
    WIN,
    TIE,
    LOSE;

    /**
     * Converts a string value to enum
     * @param gameScoreStatus string enum name
     * @return enum converted from string
     */
    public static GameScoreStatus getStatusFromString(String gameScoreStatus)
    {
        if(gameScoreStatus.toLowerCase().equals(GameScoreStatus.WIN.name().toLowerCase()))
            return GameScoreStatus.WIN;
        else if(gameScoreStatus.toLowerCase().equals(GameScoreStatus.TIE.name().toLowerCase()))
            return GameScoreStatus.TIE;
        else
            return GameScoreStatus.LOSE;
    }
}

File: app/src/main/java/com/alphadevelopmentsolutions/frcscout/Enums/GameScoreStatus.kt
package com.alphadevelopmentsolutions.frcscout.Enums

enum class GameScoreStatus
{
    WIN,
    TIE,
    LOSE;


    companion object
    {

        /**
         * Converts a string value to enum
         * @param gameScoreStatus string enum name
         * @return enum converted from string
         */
        fun getStatusFromString(gameScoreStatus: String): GameScoreStatus
        {
            return when
            {
                gameScoreStatus.toLowerCase() == GameScoreStatus.WIN.name.toLowerCase() -> WIN
                gameScoreStatus.toLowerCase() == GameScoreStatus.TIE.name.toLowerCase() -> TIE
                else -> LOSE
            }
        }
    }
}
